### DISCUSSION TITLE
Failing to install wix via dotnet tool install command

### DISCUSSION THREAD

**Discussion by TomzBench** [wixtoolset/issues#7869] [Questions] [answered]
```shell
dotnet tool install --global wix --version 4.0.3
C:\Users\tom\AppData\Local\Temp\fd3da197-62c5-4138-8823-83d9ee4d9238\restore.csproj : error NU1101: Unable to find package wix. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
The tool package could not be restored.
Tool 'wix' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
```

dotnet --version = 7.0.401

😢

**Comment by chrpai** [comment]
Works for me. I'm on 8.0 already.  What's your package sources? `dotnet nuget list source`

**Reply by TomzBench** [reply]
```sh
dotnet nuget list source
Registered Sources:
  1.  Microsoft Visual Studio Offline Packages [Enabled]
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
```
Pretty much default. dotnet newbie so my env would be typical of a default environment

**Comment by chrpai** [comment, marked as answer]
That's not typical of a default environment.  You're missing the nuget.org feed.

Registered Sources:
  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json
           
https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-add-source
dotnet nuget add source https://api.nuget.org/v3/index.json -n nuget.org

Are you a WiX newbiew as well?  Are you using Visual Studio?

**Reply by TomzBench** [reply]
Adding the source fixed the issue.

I do have what I would think to be a rational "default" Visual Studio installation which includes the .NET framework. Simply checking the .NET installation did not include the nuget.org source for me atleast.

And yes I am a Wix newbiew. I am primarily using VSCode within a flutter env and was directed to Microsoft installer framework via the flutter docs, which lead me to the wix doc after searching around for something to assist me.  

I then was following the wix getting started guide and errored on the installation.  

By "default" i mean below Visual Studio installations

![image](https://github.com/wixtoolset/issues/assets/20780378/fc29d92d-f992-4098-92dd-6e0a9f2acb40)

**Comment by chrpai** [comment]
Not sure what happened but nuget.org has always been there for me.  It'd be like having npm without registry.npmjs.org or docker without the docker hub registry.  Sure there are scenarios where that's desired but it's not the common case.

I have a FOSS project that helps getting started with WiX.  I'd be happy to spend an hour with you complimentary to help you get up to speed with WiX and MSI.   Checkout https://github.com/iswix-llc/iswix-tutorials  for a teaser.

**Reply by TomzBench** [reply]
That is very kind of you to offer.  I think I will find the tutorials helpful and will reach back to you for further assistance when closer to releasing our project.
